src/amulet_nbt/cpp/nbt_encoding/binary/read_binary.cpp

```diff
@@ -163,41 +163,46 @@ inline AmuletNBT::TagNode read_node(AmuletNBT::BinaryReader& reader, std::uint8_
 
 
 namespace AmuletNBT {
-    AmuletNBT::NamedTag read_nbt(AmuletNBT::BinaryReader& reader){
+    AmuletNBT::NamedTag read_nbt(AmuletNBT::BinaryReader& reader, bool named){
         std::uint8_t tag_id = reader.readNumeric<std::uint8_t>();
-        std::string name = read_string_tag(reader);
+        std::string name = named ? read_string_tag(reader) : "";
         AmuletNBT::TagNode node = read_node(reader, tag_id);
         return AmuletNBT::NamedTag(name, node);
     }
 
-    // Read one named tag from the string at position offset.
-    AmuletNBT::NamedTag read_nbt(const std::string& raw, std::endian endianness, AmuletNBT::StringDecode string_decode, size_t& offset){
+    // Read one (un)named tag from the string at position offset.
+    AmuletNBT::NamedTag read_nbt(const std::string& raw, std::endian endianness, AmuletNBT::StringDecode string_decode, bool named, size_t& offset){
         AmuletNBT::BinaryReader reader(raw, offset, endianness, string_decode);
-        return read_nbt(reader);
+        return read_nbt(reader, named);
+    }
+
+    // Read one (un)named tag from the string.
+    AmuletNBT::NamedTag read_nbt(const std::string& raw, std::endian endianness, AmuletNBT::StringDecode string_decode, bool named){
+        size_t offset = 0;
+        return read_nbt(raw, endianness, string_decode, named, offset);
     }
 
     // Read one named tag from the string.
     AmuletNBT::NamedTag read_nbt(const std::string& raw, std::endian endianness, AmuletNBT::StringDecode string_decode){
-        size_t offset = 0;
-        return read_nbt(raw, endianness, string_decode, offset);
+        return read_nbt(raw, endianness, string_decode, true);
```

Comment:
What has happened here?

Reply:
Oh, I guess the git diff for this is a bit confusing, it's not what it looks like.
I added a overload for `AmuletNBT::NamedTag read_nbt(const std::string& raw, std::endian endianness, AmuletNBT::StringDecode string_decode, bool named)` but without the `named` parameter. This function is used a lot (mainly in py_array_tag.cpp), so I thought I'd add this for backwards-compatibility with old C++ code. (otherwise I'd have to add a `true` parameter in every call site of this function)

Reply:
(Sorry for slow reply btw, I don't know why github aren't sending me email notifications right now)

Reply:
Ah right makes sense.